### PR TITLE
feat(es-dev-server): add es-dev-server

### PR DESCRIPTION
**Pull request by LarsDenBakker**
Rewrite of `owc-dev-server`, rebranding it as `es-dev-server` as it is not web component specific and solves a more general problem.

Changes:
- Use koa instead of express, which is simpler and more light weight
- Add actual tests :)
- Add HTTP2 flag
- Add content-based caching, unchanged files are cached
- Make node-resolve a flag which requires explicit opt-in, in preparation for import maps which won't need this flag anymore
- Add compatibility mode, which enabled using es modules, import maps etc. all the way to IE11 and makes it possible `es-dev-server` for everything you need.

For details check the readme.

### CODE REVIEW

**Comment by LarsDenBakker** [comment]
There are some items left to do, but it would be great if people can test it out and give feedback. Also please review the readme and see what can be improved.

In the `package.json` you can see different commands which execute the server with different flags.

File: packages/es-dev-server/TODO.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - inline module
+  - no external module found
+- custom SSL key / cert
+- ensure no commonjs, amd etc.
+
+Later:
+- make polyfills configurable
+- format load script
+- support typescript somehow
+- look into babel specificity: js config overrides babelrc?
```
Comment (daKmoR) [inline packages/es-dev-server/TODO.md]: if you are talking about `babel.config.js` then it's a feature
see https://babeljs.io/docs/en/config-files#project-wide-configuration

File: packages/es-dev-server/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+### Development help
+| name          |  type          | description                                                              |
+| ------------- | ---------------| ------------------------------------------------------------------------ |
+| watch         | boolean        | Reload the browser when files are edited                                 |
+| http2         | number         | Serve files over HTTP2. Sets up HTTPS with self-signed certificates      |
+
+### Code transformation
+| name          |  type          | description                                                              |
+| ------------- | ---------------| ------------------------------------------------------------------------ |
+| compatibility | number         | Compatibility mode for older browsers. Can be: `esm`, `modern` or `all`  |
```
Comment (thepassle) [inline packages/es-dev-server/README.md]: should type be string?

Comment (LarsDenBakker) [reply packages/es-dev-server/README.md]: Yes, thanks!

File: packages/es-dev-server/TODO.md (hunk 1 above)
Comment (LarsDenBakker) [reply packages/es-dev-server/TODO.md]: Not entirely. I'm calling babel from javascript using a config. I'm wondering if the user's own configuration will overwrite that, or if I will overwrite the user's configuration.

**Comment by LarsDenBakker** [comment]
All TODOs are resolved and CI is passing

File: packages/building-utils/index-html/extract-resources.js [hunk 3]
```diff
@@ -34,6 +34,8 @@ function extractResources(htmlString) {
     if (src) {
       jsModules.push(src);
       remove(moduleScript);
+    } else {
+      inlineModules.push(moduleScript);
```
Comment (daKmoR) [inline packages/building-utils/index-html/extract-resources.js]: shouldn't inline module also be removed?

File: packages/webpack-index-html-plugin/test/snapshots/external-loader/index.html [hunk 4]
```diff
@@ -1 +1,5 @@
-<html lang="en-GB"><head><title>My app</title><style>my-app{display:block}</style><link rel="preload" href="app.889b20bcbef01fce0e59.js" as="script"></head><body><h1><span>Hello world!</span></h1><my-app></my-app><script>console.log("hello inline script");</script><script src="polyfills/core-js.79b2367fd020fda0c519294df7db91bf.js" nomodule=""></script><script src="loader.7dfe0d142b9de5fec887b82320688498.js"></script></body></html>
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (daKmoR) [inline packages/webpack-index-html-plugin/test/snapshots/external-loader/index.html]: merge conflict?

**Comment by Pupix** [comment]
Just cloned this branch to test out https://github.com/open-wc/open-wc/issues/507 and it seems fixed.

File: packages/building-utils/index-html/extract-resources.js (hunk 3 above)
Comment (LarsDenBakker) [reply packages/building-utils/index-html/extract-resources.js]: Yes, I'll add it

File: .gitignore [hunk 5]
```diff
@@ -10,6 +10,7 @@ coverage/
 
 ## npm
 node_modules
+!packages/es-dev-server/test/**/node_modules
```
[the remark this answers is not in the document]
Comment (LarsDenBakker) [reply .gitignore]: Good idea I'll add that

**Pull request merged**